### PR TITLE
Split JS utility functions and cleanup form layout templates

### PR DESCRIPTION
**Pull request by vidya-ram**
1. Move JS functions out of  form layout templates.
2. Split JS utility functions to multiple helper files to reduce imported file sizes and duplicate imports.
3. For Submission page, load code mirror asynchronously to improve page performance.

### CODE REVIEW

File: funnel/templates/submission_admin_panel.html.jinja2 [hunk 1]
```diff
@@ -43,6 +37,12 @@
       {{ rendersubmit([("add-label", "Save", '')]) }}
     </form>
   {% endif %}
+  {%- if proposal_move_form %}
```
Comment (vidya-ram) [inline funnel/templates/submission_admin_panel.html.jinja2:40]: Moved these forms down for better visual hierarchy

File: funnel/templates/submission.html.jinja2 [hunk 2]
```diff
@@ -283,11 +277,11 @@
         user: {% if current_auth.user -%}{{ { 'fullname': current_auth.user.fullname, 'avatar': current_auth.user.avatar, 'profile_url': current_auth.user.profile_url }|tojson }}{% else %}{}{% endif %},
         loginUrl: "{{ url_for('login') }}",
         lastSeenUrl: {% if proposal.commentset.current_roles.document_subscriber %}{{ proposal.commentset.url_for('update_last_seen_at')|tojson }}{% else %}false{% endif %},
+        codemirrorUrl: {% assets "js_codemirrormarkdown" -%}{{ ASSET_URL|tojson }}{%- endassets -%}
```
Comment (vidya-ram) [inline funnel/templates/submission.html.jinja2:280]: Codemirror is used for the comments form in submission page. Loading it asynchronously to improve page performance.

File: funnel/templates/project.html.jinja2 [hunk 3]
```diff
@@ -206,20 +206,20 @@
 {% endblock %}
 
 {% block footerinnerscripts %}
-  <script src="{{ built_asset('project.js') }}" type="text/javascript"></script>
-  <script type="text/javascript">
-    $(function() {
-      var projectConfig = {};
+  {%- if project.primary_venue.has_coordinates -%}
```
Comment (vidya-ram) [inline funnel/templates/project.html.jinja2:209]: `projectInit` is for initialising map. To improve page performance, load leaflet assets if the project has a venue.

File: funnel/templates/formlayout.html.jinja2 [hunk 4]
```diff
@@ -28,96 +28,16 @@
...
-          $(window).bind('beforeunload', function() {
-            if(haveDirtyFields()){
-              return 'You have unsaved changes on this page. Do you want to leave this page?';
-            }
-          });
-        }
-      });
-    </script>
-  {%- endif %}
+  {% block innerscripts %}{% endblock %}
```
Comment (vidya-ram) [inline funnel/templates/formlayout.html.jinja2:42]: There are other form pages that extend formlayout and this block is for page related scripts.

**Review by jace — APPROVED** [review]

File: funnel/assets/js/utils/embedvideo.js [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    if (type === 'youtube') {
+      videoEmbedUrl = `<iframe src='//www.youtube.com/embed/${videoId}' frameborder='0' allowfullscreen></iframe>`;
+    } else if (type === 'vimeo') {
+      videoEmbedUrl = `<iframe src='https://player.vimeo.com/video/${videoId}' frameborder='0' allowfullscreen></iframe>`;
+    }
+    if (videoEmbedUrl) {
+      videoWrapper.innerHTML = videoEmbedUrl;
+    }
+  },
+};
```
[the remark this answers is not in the document]
Comment (vidya-ram) [reply funnel/assets/js/utils/embedvideo.js:45]: Livestream urls use this. It is not a Video model yet.

File: funnel/static/sass/baseframe-material/_tabs.scss [hunk 6]
```diff
@@ -130,6 +132,11 @@
   }
 }
 
+/* Hide scrollbar for Chrome, Safari and Opera */
+.tabs::-webkit-scrollbar {
+  display: none;
+}
```
[the remark this answers is not in the document]
Comment (vidya-ram) [reply funnel/static/sass/baseframe-material/_tabs.scss:138]: The tabs on the accounts page uses horizontal scroll and the scrollbar is overlapping the content

**Pull request merged**